src/doc/book/traits.md

```diff
@@ -243,27 +243,7 @@ to know more about [operator traits][operators-and-overloading].
 # Rules for implementing traits
 
 So far, we’ve only added trait implementations to structs, but you can
-implement a trait for any type. So technically, we _could_ implement `HasArea`
-for `i32`:
-
-```rust
-trait HasArea {
-    fn area(&self) -> f64;
-}
-
-impl HasArea for i32 {
-    fn area(&self) -> f64 {
-        println!("this is silly");
-
-        *self as f64
-    }
-}
-
-5.area();
-```
-
-It is considered poor style to implement methods on such primitive types, even
-though it is possible.
+implement a trait for any type such as `i32`.
```

Comment:
That would actually be bad style (imho). One cannot implement a meaningful area for `i32`, only for the newtype `struct Square(i32)`. This is why I wanted to remove the example altogether because that possibly  lengthy discussion is out of scope of that particular paragraph.